pkg/tools/serialize/serialize.go

```diff
@@ -44,25 +40,28 @@ func NewExecutor() Executor {
 // AsyncExec - guarantees f() will be executed Exclusively and in the Order submitted.
 //        It immediately returns a channel that will be closed when f() has completed execution.
 func (e *Executor) AsyncExec(f func()) <-chan struct{} {
-	// Initialize *once*
-	e.init.Do(func() {
-		e.execCh = make(chan func(), channelSize)
+	done := make(chan struct{})
+	e.mutex.Lock()
+	e.queue = append(e.queue, func() {
```

Comment:
Apropos of @haiodo 's comment about benchmarking... append here is a rather expensive operation.

Reply:
@edwarnicke please take a look at the benchmark result https://github.com/networkservicemesh/sdk/pull/519#issuecomment-707750806

Reply:
slices in go are optimized